src/octree/octree_iterator.rs

```diff
@@ -130,3 +130,42 @@ pub fn intersecting_node_ids(
     }
     node_ids
 }
+
+pub struct NodeIdIterator<'a> {
+    octree: &'a Octree,
+    filter_func: Box<Fn(&NodeId, &Octree) -> bool + 'a>,
```

Comment:
My first preference would be to integrate this directly with PointCulling instead of functions, and to use the more optimized distinction between Inside, Crossing and Outside to automatically add child nodes. If we use functions, how about making this a template parameter instead of a trait object?

Reply:
As mentioned offline, the iterator is more generic than the filtering of the point culling. This distinction occurs for Frustum and a specific method returns the node ids.

Reply:
Yes, but can we make this a parameter of the struct, I.e. `struct NodeIdsIterator<F>` and `filter_func: F`, and restrict it to `Fn(&NodeId, &Octree) -> bool` in the impl?

Reply:
Actually, seems like you can directly embed the trait bound into the struct definition (`where F: Fn(&NodeId, &Octree) -> bool`)

Reply:
done